File: java/eyes.playwright.java/src/main/java/com/applitools/eyes/playwright/fluent/IPlaywrightCheckSettings.java
package com.applitools.eyes.playwright.fluent;

import com.applitools.ICheckSettings;
import com.applitools.eyes.AccessibilityRegionType;
import com.applitools.eyes.Padding;
import com.applitools.eyes.Region;
import com.microsoft.playwright.ElementHandle;
import com.microsoft.playwright.Frame;
import com.microsoft.playwright.Locator;

public interface IPlaywrightCheckSettings extends ICheckSettings {

    PlaywrightCheckSettings region(Region region);

    PlaywrightCheckSettings region(String selector);

    PlaywrightCheckSettings region(Locator locator);

    PlaywrightCheckSettings region(ElementHandle elementHandle);

    PlaywrightCheckSettings frame(String frameNameOrId);

    PlaywrightCheckSettings frame(Integer frameIndex);

    PlaywrightCheckSettings frame(Locator locator);

    PlaywrightCheckSettings frame(ElementHandle elementHandle);

    PlaywrightCheckSettings ignore(String selector);

    PlaywrightCheckSettings ignore(Locator locator);

    PlaywrightCheckSettings ignore(ElementHandle elementHandle);

    PlaywrightCheckSettings ignore(String selector, String regionId);

    PlaywrightCheckSettings ignore(Locator locator, String regionId);

    PlaywrightCheckSettings ignore(ElementHandle elementHandle, String regionId);

    PlaywrightCheckSettings ignore(String selector, Padding padding);

    PlaywrightCheckSettings ignore(Locator locator, Padding padding);

    PlaywrightCheckSettings ignore(ElementHandle elementHandle, Padding padding);

    PlaywrightCheckSettings layout(String selector);

    PlaywrightCheckSettings layout(Locator locator);

    PlaywrightCheckSettings layout(ElementHandle elementHandle);

    PlaywrightCheckSettings layout(String selector, String regionId);

    PlaywrightCheckSettings layout(Locator locator, String regionId);

    PlaywrightCheckSettings layout(ElementHandle elementHandle, String regionId);

    PlaywrightCheckSettings layout(String selector, Padding padding);

    PlaywrightCheckSettings layout(Locator locator, Padding padding);

    PlaywrightCheckSettings layout(ElementHandle elementHandle, Padding padding);

    PlaywrightCheckSettings strict(String selector);

    PlaywrightCheckSettings strict(Locator locator);

    PlaywrightCheckSettings strict(ElementHandle elementHandle);

    PlaywrightCheckSettings strict(String selector, String regionId);

    PlaywrightCheckSettings strict(Locator locator, String regionId);

    PlaywrightCheckSettings strict(ElementHandle elementHandle, String regionId);

    PlaywrightCheckSettings strict(String selector, Padding padding);

    PlaywrightCheckSettings strict(Locator locator, Padding padding);

    PlaywrightCheckSettings strict(ElementHandle elementHandle, Padding padding);

    PlaywrightCheckSettings content(String selector);

    PlaywrightCheckSettings content(Locator locator);

    PlaywrightCheckSettings content(ElementHandle elementHandle);

    PlaywrightCheckSettings content(String selector, String regionId);

    PlaywrightCheckSettings content(Locator locator, String regionId);

    PlaywrightCheckSettings content(ElementHandle elementHandle, String regionId);

    PlaywrightCheckSettings content(String selector, Padding padding);

    PlaywrightCheckSettings content(Locator locator, Padding padding);

    PlaywrightCheckSettings content(ElementHandle elementHandle, Padding padding);

    PlaywrightCheckSettings ignoreColors();

    PlaywrightCheckSettings ignoreColors(String selector);

    PlaywrightCheckSettings ignoreColors(Locator locator);

    PlaywrightCheckSettings ignoreColors(ElementHandle elementHandle);

    PlaywrightCheckSettings ignoreColors(String selector, String regionId);

    PlaywrightCheckSettings ignoreColors(Locator locator, String regionId);

    PlaywrightCheckSettings ignoreColors(ElementHandle elementHandle, String regionId);

    PlaywrightCheckSettings ignoreColors(String selector, Padding padding);

    PlaywrightCheckSettings ignoreColors(Locator locator, Padding padding);

    PlaywrightCheckSettings ignoreColors(ElementHandle elementHandle, Padding padding);

    PlaywrightCheckSettings floating(String selector, int maxUpOffset, int maxDownOffset, int maxLeftOffset, int maxRightOffset);

    PlaywrightCheckSettings floating(Locator locator, int maxUpOffset, int maxDownOffset, int maxLeftOffset, int maxRightOffset);

    PlaywrightCheckSettings floating(ElementHandle elementHandle, int maxUpOffset, int maxDownOffset, int maxLeftOffset, int maxRightOffset);

    PlaywrightCheckSettings floating(int maxOffset, String selector);

    PlaywrightCheckSettings floating(int maxOffset, Locator locator);

    PlaywrightCheckSettings floating(int maxOffset, ElementHandle elementHandle);

    PlaywrightCheckSettings accessibility(String selector, AccessibilityRegionType type);

    PlaywrightCheckSettings accessibility(Locator locator, AccessibilityRegionType type);

    PlaywrightCheckSettings accessibility(ElementHandle elementHandle, AccessibilityRegionType type);

    PlaywrightCheckSettings scrollRootElement(String selector);

    PlaywrightCheckSettings scrollRootElement(Locator locator);

    PlaywrightCheckSettings scrollRootElement(ElementHandle elementHandle);

    PlaywrightCheckSettings layoutBreakpoints(int... breakpoints);

    PlaywrightCheckSettings layoutBreakpoints(Boolean shouldSet);
}

File: dotnet/Eyes.Playwright.DotNet/Fluent/IPlaywrightCheckSettings.cs
using System;
using Applitools.Utils.Geometry;
using Microsoft.Playwright;

namespace Applitools.Playwright.Fluent
{
    public interface IPlaywrightCheckSettings : ICheckSettings
    {

        PlaywrightCheckSettings Region(Region region);

        PlaywrightCheckSettings Region(string selector);

        PlaywrightCheckSettings Region(ILocator locator);

        PlaywrightCheckSettings Region(IElementHandle elementHandle);

        PlaywrightCheckSettings Frame(string frameNameOrId);

        PlaywrightCheckSettings Frame(int frameIndex);

        PlaywrightCheckSettings Frame(ILocator locator);

        PlaywrightCheckSettings Frame(IElementHandle elementHandle);

        PlaywrightCheckSettings Ignore(string selector);

        PlaywrightCheckSettings Ignore(ILocator locator);

        PlaywrightCheckSettings Ignore(IElementHandle elementHandle);

        PlaywrightCheckSettings Ignore(string selector, string regionId);

        PlaywrightCheckSettings Ignore(ILocator locator, string regionId);

        PlaywrightCheckSettings Ignore(IElementHandle elementHandle, string regionId);

        PlaywrightCheckSettings Ignore(string selector, Padding padding);

        PlaywrightCheckSettings Ignore(ILocator locator, Padding padding);

        PlaywrightCheckSettings Ignore(IElementHandle elementHandle, Padding padding);

        PlaywrightCheckSettings Layout(string selector);

        PlaywrightCheckSettings Layout(ILocator locator);

        PlaywrightCheckSettings Layout(IElementHandle elementHandle);

        PlaywrightCheckSettings Layout(string selector, string regionId);

        PlaywrightCheckSettings Layout(ILocator locator, string regionId);

        PlaywrightCheckSettings Layout(IElementHandle elementHandle, string regionId);

        PlaywrightCheckSettings Layout(string selector, Padding padding);

        PlaywrightCheckSettings Layout(ILocator locator, Padding padding);

        PlaywrightCheckSettings Layout(IElementHandle elementHandle, Padding padding);

        PlaywrightCheckSettings Strict(string selector);

        PlaywrightCheckSettings Strict(ILocator locator);

        PlaywrightCheckSettings Strict(IElementHandle elementHandle);

        PlaywrightCheckSettings Strict(string selector, string regionId);

        PlaywrightCheckSettings Strict(ILocator locator, string regionId);

        PlaywrightCheckSettings Strict(IElementHandle elementHandle, string regionId);

        PlaywrightCheckSettings Strict(string selector, Padding padding);

        PlaywrightCheckSettings Strict(ILocator locator, Padding padding);

        PlaywrightCheckSettings Strict(IElementHandle elementHandle, Padding padding);

        [Obsolete("Use IgnoreColors")]
        PlaywrightCheckSettings Content(string selector);

        [Obsolete("Use IgnoreColors")]
        PlaywrightCheckSettings Content(ILocator locator);

        [Obsolete("Use IgnoreColors")]
        PlaywrightCheckSettings Content(IElementHandle elementHandle);

        [Obsolete("Use IgnoreColors")]
        PlaywrightCheckSettings Content(string selector, string regionId);

        [Obsolete("Use IgnoreColors")]
        PlaywrightCheckSettings Content(ILocator locator, string regionId);

        [Obsolete("Use IgnoreColors")]
        PlaywrightCheckSettings Content(IElementHandle elementHandle, string regionId);

        [Obsolete("Use IgnoreColors")]
        PlaywrightCheckSettings Content(Region[] regions);

        [Obsolete("Use IgnoreColors")]
        PlaywrightCheckSettings Content(Region region, params Region[] regions);
        
        [Obsolete("Use IgnoreColors")]
        PlaywrightCheckSettings Content(string selector, Padding padding);

        [Obsolete("Use IgnoreColors")]
        PlaywrightCheckSettings Content(ILocator locator, Padding padding);

        [Obsolete("Use IgnoreColors")]
        PlaywrightCheckSettings Content(IElementHandle elementHandle, Padding padding);

        PlaywrightCheckSettings IgnoreColors(string selector);

        PlaywrightCheckSettings IgnoreColors(ILocator locator);

        PlaywrightCheckSettings IgnoreColors(IElementHandle elementHandle);

        PlaywrightCheckSettings IgnoreColors(string selector, string regionId);

        PlaywrightCheckSettings IgnoreColors(ILocator locator, string regionId);

        PlaywrightCheckSettings IgnoreColors(IElementHandle elementHandle, string regionId);

        PlaywrightCheckSettings IgnoreColors(string selector, Padding padding);

        PlaywrightCheckSettings IgnoreColors(ILocator locator, Padding padding);

        PlaywrightCheckSettings IgnoreColors(IElementHandle elementHandle, Padding padding);

        PlaywrightCheckSettings IgnoreColors(Region[] regions);
        
        PlaywrightCheckSettings IgnoreColors(Region region, params Region[] regions);
        
        PlaywrightCheckSettings Floating(string selector, int maxUpOffset, int maxDownOffset, int maxLeftOffset,
            int maxRightOffset);

        PlaywrightCheckSettings Floating(ILocator locator, int maxUpOffset, int maxDownOffset, int maxLeftOffset,
            int maxRightOffset);

        PlaywrightCheckSettings Floating(IElementHandle elementHandle, int maxUpOffset, int maxDownOffset,
            int maxLeftOffset, int maxRightOffset);

        PlaywrightCheckSettings Floating(int maxOffset, string selector);

        PlaywrightCheckSettings Floating(int maxOffset, ILocator locator);

        PlaywrightCheckSettings Floating(int maxOffset, IElementHandle elementHandle);

        PlaywrightCheckSettings Accessibility(string selector, AccessibilityRegionType type);

        PlaywrightCheckSettings Accessibility(ILocator locator, AccessibilityRegionType type);

        PlaywrightCheckSettings Accessibility(IElementHandle elementHandle, AccessibilityRegionType type);

        PlaywrightCheckSettings ScrollRootElement(string selector);

        PlaywrightCheckSettings ScrollRootElement(ILocator locator);

        PlaywrightCheckSettings ScrollRootElement(IElementHandle elementHandle);

        PlaywrightCheckSettings LayoutBreakpoints(params int[] breakpoints);

        PlaywrightCheckSettings LayoutBreakpoints(bool shouldSet);
    }
}